islandora_solr_config/includes/rss_results.inc

```diff
@@ -191,7 +190,85 @@ class IslandoraSolrResultsRSS extends IslandoraSolrResults {
     $result['url'] = $base_url;
     $result['description'] = t('Aggregated search results of: @query', array('@query' => $query));
     $result['langcode'] = NULL;
+    $feed_url = request_uri();
+    $page_params = drupal_get_query_parameters();
+
+    if (!isset($page_params['page'])) {
+      $page_number = '0';
+    }
+    else {
+      $page_number = $page_params['page'];
+    }
+    if (isset($page_params['page'])) {
+      $first_params = $page_params;
+      unset($first_params['page']);
+      $link_first = url(
+        request_path(),
+        array(
+          'absolute' => TRUE,
+          'query' => $first_params,
+        )
+      );
+    }
+    else {
+      $link_first = $base_url . $feed_url;
+    }
+
+    $page_next = $page_number + 1;
+    if ($page_number > 0) {
+      $page_prev = $page_number - 1;
+    }
+
+    $next_page = array(
+      'query' => array(
+        'page' => $page_next,
+      )+$page_params,
+    );
+
+    // Only set Next link if there is a next page.
+    $rss_limit = variable_get('islandora_solr_config_rss_limit', 50);
+    if (count($docs) == $rss_limit) {
+      $link_next = url($base_url . "/" . request_path(), $next_page);
+    }
+
+    if (isset($page_prev)) {
+      if ($page_prev > 0) {
+        $prev_page = array(
+          'query' => array(
+            'page' => $page_prev,
+          )+$page_params,
+        );
+        $link_prev = url($base_url . "/" . request_path(), $prev_page);
+      }
+      else {
+        $link_prev = $link_first;
+      }
+    }
+    else {
+      $link_prev = "";
```

Comment:
What is this trying to do? I understand is as: If there is no page passed to the search than the previous page is this page. In that case would it make sense to instead just not include the the previous page link at all? (Falling back to the old behaviour).